### PR TITLE
SNOW-1665775: Move resample.py into src/…/modin/plugin/extensions/

### PR DESCRIPTION
**Pull request by sfc-gh-helmeleegy**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1665775

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Move resample.py into src/…/modin/plugin/extensions/.

### CODE REVIEW

File: docs/source/modin/resampling.rst [hunk 1]
```diff
@@ -2,7 +2,7 @@
 Resampling
 =============================
 
-.. currentmodule:: snowflake.snowpark.modin.pandas.resample
+.. currentmodule:: modin.pandas
```
Comment (sfc-gh-joshi) [inline docs/source/modin/resampling.rst:5]: We still specify `snowflake.snowpark.modin.plugin.extensions.groupby_overides` for GroupBy, can you check if the documentation build here is still using Snowpark pandas docs?

**Review by sfc-gh-joshi — CHANGES_REQUESTED** [review]
one question about docs

File: docs/source/modin/resampling.rst (hunk 1 above)
Comment (sfc-gh-helmeleegy) [reply docs/source/modin/resampling.rst]: I switched it to `snowflake.snowpark.modin.plugin.extensions.resample_overrides`

**Review by sfc-gh-joshi — APPROVED** [review]

File: src/snowflake/snowpark/modin/plugin/_internal/telemetry.py [hunk 2]
```diff
@@ -543,7 +543,7 @@ def __new__(
         cls, name: str, bases: tuple, attrs: dict[str, Any]
     ) -> Union[
         "snowflake.snowpark.modin.plugin.extensions.groupby_overrides.DataFrameGroupBy",
-        "snowflake.snowpark.modin.pandas.resample.Resampler",
```
[the remark this answers is not in the document]
Comment (sfc-gh-helmeleegy) [reply src/snowflake/snowpark/modin/plugin/_internal/telemetry.py:546]: I think we already have one:
https://github.com/snowflakedb/snowpark-python/blob/722701149fba5c6cea2aad3a42c46f17da528b9c/tests/integ/modin/test_telemetry.py#L343
Right?

**Review by sfc-gh-azhan — APPROVED** [review]

**Pull request merged**